src/backend/scraping.go

```
@@ -0,0 +1,169 @@
+		page, err := scrapeWikipedia(url, lang)
+		if err == nil && page.Title != "" {
+			return page, lang, nil
+		}
+		log.Printf("Failed scraping %s (%s): %v", term, lang, err)
+	}
+	return Page{}, "", fmt.Errorf("no valid Wikipedia page found for term '%s'", term)
+}
+
+func buildWikipediaURL(term, lang string) string {
+	term = strings.ReplaceAll(term, " ", "_")
+	return fmt.Sprintf("https://%s.wikipedia.org/wiki/%s", lang, strings.Title(term))
```

golangci-lint (1.64.8)

SA1019: strings.Title has been deprecated since Go 1.18 and an alternative has been available since Go 1.0: The rule Title uses for word boundaries does not handle Unicode punctuation properly. Use golang.org/x/text/cases instead. 

(staticcheck)